Sources/SwiftCompilerSupport/ConsistencyCheck.swift

```diff
@@ -39,7 +39,9 @@ extension Syntax {
 @_spi(SwiftCompiler)
 public func _parserConsistencyCheck(
   bufferPtr: UnsafePointer<UInt8>, bufferLength: Int,
-  filename: UnsafePointer<UInt8>, flags: CUnsignedInt
+  filename: UnsafePointer<UInt8>, flags: CUnsignedInt,
+  ctx: OpaquePointer,
+  hook: @convention(c) (Int, UnsafePointer<Int8>, OpaquePointer) -> Void
```

Comment:
Could you add doc comments for `ctx` and `hook`, including what the parameters to `hook` are? Also, I think `hook` would probably be better named `report diagnostic` or something like that.

Reply:
Sure, I'll do it on the C prototype though.